src/nvim/eval.c

```diff
@@ -16365,6 +16373,26 @@ void set_vim_var_list(int idx, list_T *val)
     ++val->lv_refcount;
 }
 
+/// Set Dictionary v: variable to "val".
+void set_vim_var_dict(int idx, dict_T *val)
+{
+  dict_unref(vimvars[idx].vv_dict);
+
+  // Set readonly
+  int todo = (int)val->dv_hashtab.ht_used;
+  for (hashitem_T *hi = val->dv_hashtab.ht_array; todo > 0 ; ++hi) {
+    if (!HASHITEM_EMPTY(hi)) {
+      --todo;
+      HI2DI(hi)->di_flags = DI_FLAGS_RO | DI_FLAGS_FIX;
+    }
```

Comment:
@Shougo My mistake, I should have looked first. I see now that you are using the pattern used commonly in the codebase.